src/astroid.cc

```diff
@@ -158,6 +163,16 @@ namespace Astroid {
     if (vm.count ("disable-log")) {
       LOG (warn) << "disabling log";
       logging::core::get()->set_logging_enabled (false);
+    } else if (vm.count ("log-level")) {
+      ustring llevel = vm["log-level"].as<ustring> ();
+
+      /* Not very elegant map for string to enum */
+      std::map<std::string, logging::trivial::severity_level> sevmap = boost::assign::map_list_of
+        ("trace", logging::trivial::trace)("debug", logging::trivial::debug)
+        ("info", logging::trivial::info)("warning", logging::trivial::warning)
+        ("error", logging::trivial::error)("fatal", logging::trivial::fatal);
```

Comment:
Is it possible to re-write this to only use standard C++14 or later? It's best to use Boost as little as possible, then we avoid the extra `include` as well.

Reply:
probably possible, this was just the easiest I could come up with.

out of curiosity, why use boost as little as possible, isn't the goal of a library to be used. Is there a severe penalty involved?